client/packages/common/src/ui/components/modals/ConfirmationModal/ConfirmationModalContext.ts

```diff
@@ -9,7 +9,9 @@ export interface ConfirmationModalState {
   iconType?: IconType;
   buttonLabel?: string | undefined;
   cancelButtonLabel?: string | undefined;
-  onConfirm?: (() => void) | (() => Promise<void>);
+  onConfirm?:
+    | ((state: ConfirmationModalState) => void)
+    | ((state: ConfirmationModalState) => Promise<void>);
```

Comment:
Give the onConfirm call access to the rest of the confirmation state if needed - used this to pass "onCancel" through to the subsequent modal..